include/maliput/test_utilities/maliput_types_compare.h

```diff
@@ -37,57 +37,68 @@ namespace maliput {
 namespace api {
 namespace test {
 
-// Compares equality within @p tolerance deviation of two InertialPosition objects.
-// @param pos1 A InertialPosition object to compare.
-// @param pos2 A InertialPosition object to compare.
-// @param tolerance An allowable absolute deviation for each InertialPosition's
-// coordinate.
-// @return ::testing::AssertionFailure() When InertialPosition objects are different.
-// @return ::testing::AssertionSuccess() When InertialPosition objects are within
-// the @p tolerance deviation.
+/// Asserts equality within @p tolerance of the maliput::api::LanePositionResult @p lpr_a and @p lpr_B.
+/// @param lpr_a The first LanePositionResult to compare.
+/// @param lpr_b The second LanePositionResult to compare.
+/// @param tolerance The tolerance to use for the comparison.
+#define IsLanePositionResultClose(lpr_a, lpr_b, tolerance)                                           \
+  do {                                                                                               \
+    EXPECT_TRUE(IsLanePositionClose(lpr_a.lane_position, lpr_b.lane_position, tolerance));           \
+    EXPECT_TRUE(IsInertialPositionClose(lpr_a.nearest_position, lpr_b.nearest_position, tolerance)); \
+    EXPECT_NEAR(lpr_a.distance, lpr_b.distance, tolerance);                                          \
+  } while (0);
```

Comment:
Noe that it is here, we should build a function and return an assertion result with the result of the comparison. Instead of a macro.

Reply:
Makes sense!

Reply:
Done